### ISSUE TITLE
RFD - Set pinned dependencies used by Terraform in `constants.py`

### ISSUE THREAD

**Issue by iameskild** [nebari-dev/governance#20]

**Comment by viniciusdc** [comment]
I agree with the pinning version, mainly because left and then unpinned always cause trouble with old releases as well, then migrating it will require hotfixes or extra custom changes. The only point I made, and noticed that @iameskild commented in the RFD above as well, is that we will need to think very well about how we will keep the updates of these pins regularly.

> As part of this RFD, I propose we review, test, and upgrade our dependencies once per quarter as part of our release process.

This also seems reasonable to me, though depending on the changes a new version might grant once unpinned, stabilizing the branch while testing/releasing could make us lose the release schedule (depending on the affected resources). So instead, I would suggest that after the last release for a quester, the first task after releasing will be unpinning and testing stability, then basically all releases in the start of the quarter will have the new pins as well as will be stable already (not affecting the release process)

**Comment by iameskild** [comment]
Thanks a lot for the feedback @viniciusdc :)

I agree with you that we should start dependency testing as early as possible for the subsequent release. This would give us time to adjust should we need to upgrade one of the dependencies.

**Comment by trallard** [comment]
Since this has been open for a while I am moving this RFD towards decision/voting.
We are using a [consent decision-making process](https://www.sociocracyforall.org/consent-decision-making/) for RFD approval therefore:

- [ ] 07-08 Feb 2023: Read through the proposal and ask any clarifying questions (everyone makes sure to read and understand the proposal and ask any questions needed to understand this better). Also, note your overall feeling about the proposal (i.e. support, do not support)
- [ ] 07-08 Feb 2023: Allow @iameskild  as the author to reply to questions and make any amendments
- [ ] 09-10 Feb 2023: decision-making round: @trallard  will kick off the consent decision-making round, integrate objections and resolve objections
- [ ] 13 Feb 2023: Ratify decision

cc @nebari-dev/maintainers

**Comment by iameskild** [comment]
RFD updated today based on discussion and notes from Tuesday's Nebari community call, [see notes](https://hackmd.io/@nebari/HkIyLID1h).

**Comment by Adam-D-Lewis** [comment]
I'm cleaning up the RFD's and I'm going to close this for now due to no activity, but feel free to re-open if needed!
